### DISCUSSION TITLE
VSCode: how to toggle copilot on/off without questions?

### DISCUSSION THREAD

**Discussion by dandavison** [community/community#7553] [Copilot Conversations]
`github.copilot.toggleCopilot` generates a notification asking whether to toggle it globally or just for the current language. I'd like to make that choice once (or have it made for me) and thus have a quick toggle with no questions asked. How do I do that?

**Comment by miguelsolorio** [comment]
Made the same request in https://github.com/github/feedback/discussions/7605, would love to avoid the need for this notification:

![image](https://user-images.githubusercontent.com/35271042/141359711-e74449f9-13a6-4bfd-8f9c-4dbc54d2b2c1.png)

So having an option or setting to toggle globally would be my preference.

**Comment by dandavison** [comment]
Until the extension gets rid of the global/local question on toggle, I'm thinking we can use `github.copilot.inlineSuggest.enable` for this. E.g. with the [settings-cycler](https://marketplace.visualstudio.com/items?itemName=hoovercj.vscode-settings-cycler) extension, then the following creates a toggle that seems to work:

```json
    {
      "id": "copilot",
      "overrideWorkspaceSettings": true,
      "values": [
        {
          "github.copilot.inlineSuggest.enable": false
        },
        {
          "github.copilot.inlineSuggest.enable": true
        }
      ]
    },
```

```json
  {
    "key": "ctrl+x p",
    "command": "settings.cycle.copilot",
    "when": "editorFocus"
  },
```

**Reply by ATSiem** [reply]
I disabled inlineSuggest, and call Copilot with a keybinding. I prefer this, to the suggestions which are often irrelevant and distracting.

However, seeing the second pane with 10 code suggestions when I need them by pressing "command+enter", that's wonderful.

<img width="587" alt="image" src="https://user-images.githubusercontent.com/1068543/155060551-1a557cbc-9c46-4707-aa40-be4fb36bb764.png">

**Reply by agusterodin** [reply]
I like this a lot. I just wish there was still a notification to tell you if you just enabled or disabled it (no clicking required).

**Comment by AJTJ** [comment]
The issue with opening a second pane is that your cursor remains in the second tab area if you already had a tab open there. The ideal solution would be one where we can actively disable/enable inline suggestions. Currently there is no good solution for this, which seems like a massive oversight.

**Comment by JohnnyBill** [comment]
Would be nice to have a fix (without workarounds) for this issue soon.
Sometimes Copilot is quite distracting and it would be nice to be able to toggle it with a single keystroke.
Overall, off course, its an awesome tool :)

**Comment by mattstam** [comment]
Being able to quickly enable/disable Copilot without the popup would be awesome.

Until then, does anyone know of a keybinding to actually highlight the popup and buttons without using a mouse?

**Comment by jmaha** [comment]
Great suggestions. I too find Copilot's inline suggestions frequently wrong or distracting so I disable it most of the time, but it would be _way_ more useful if it was possible to quickly enable/disable with a single keystroke.

**Comment by dandavison** [comment]
Just to confirm, the solution I suggested above requires a bit of setup but does work very well: https://github.com/orgs/community/discussions/7553#discussioncomment-1629050

**Comment by Butanium** [comment]
Using the @dandavison solution:
**settings.json**
```json
"settings.cycle": [
        {
            "id": "copilot",
            "overrideWorkspaceSettings": true,
            "values": [
                {
                    "github.copilot.inlineSuggest.enable": false
                },
                {
                    "github.copilot.inlineSuggest.enable": true
                }
            ]
        }
    ]
```
**keybindings.json**
```json
 {
        "key": "ctrl+k ctrl+alt+o",
        "command": "settings.cycle.copilot",
    },
```
Works for me.  Note that it doesn't update the bottom right icon. For example, If you disable it while it was computing completion, the icon will infinitely cycle.

**Reply by zachbellay** [reply]
Thank you, I couldn't quite piece together where I was supposed to put @dandavison's very helpful suggestion, thank you for this!!

**Comment by AsafMeizner** [comment]
in vs code there is a button in the button right to quickly enable and disable copilot

**Reply by magnus-eriksson** [reply]
> in vs code there is a button in the button right to quickly enable and disable copilot

...which will show the question we're trying to avoid seeing. Like the title says: "_without_ questions".

**Reply by mattstam** [reply]
Or at least how to do so without needing to use the mouse.

**Comment by stilliard** [comment]
Would be super useful. 
In case it helps, similar to @dandavison's suggestion you can also use the [Toggle](https://marketplace.visualstudio.com/items?itemName=rebornix.toggle) extension and add the following to your `keybindings.json` file with whatever key combination you'd like:

```json
    {
        "key": "ctrl+alt+c ctrl+alt+p",
        "command": "toggle",
        "when": "editorTextFocus",
        "args": {
            "id": "copilot-enabled",
            "value": [
                {
                    "github.copilot.inlineSuggest.enable": true
                },
                {
                    "github.copilot.inlineSuggest.enable": false
                }
            ]
        }
    }
```

**Comment by slsp8752** [comment]
If anyone is trying to do this in 2025, the value to toggle is now "github.copilot.editor.enableAutoCompletions". "github.copilot.inlineSuggest.enable" has been deprecated and no longer works for me.

**Comment by batur** [comment]
Two things should be considered about it for future visitors: 

1. The `"github.copilot.inlineSuggest.enable"` looks deprecated.
2. The bottom icon vanished with Github Copilot extension updates

I using like below:
```json
{
  "key": "cmd+shift+[BracketRight]",
  "command": "toggle",
  "when": "editorTextFocus && !editorReadonly && !inSnippetMode && !hasOtherSuggestions && !suggestWidgetVisible",
  "args": {
    "id": "copilot-enabled",
    "overrideWorkspaceSettings": true,
    "value": [
      {
        "github.copilot.editor.enableAutoCompletions": true,
        "workbench.colorCustomizations": {
          "statusBar.background": "#4EC9B0",
          "statusBar.foreground": "#000000"
        }
      },
      {
        "github.copilot.editor.enableAutoCompletions": false,
        "workbench.colorCustomizations": {
          "statusBar.background": "#8B7813"
        }
      }
    ]
  }
}
```

When toggling to auto completions status bar color changes.

**Reply by techthinkervivek** [reply]
![vs code 4](https://github.com/user-attachments/assets/d1e61070-f2a5-4173-b61b-0aa1b2a31703)
i am totally disturbed by this error in vs code from more than a month , please help me

**Comment by techthinkervivek** [comment]
i am totally disturbed by this error in vs code from more than a month , please help me  , i have tried to add shortcuts through json file but none worked for me

**Comment by techthinkervivek** [comment]
I
![vs code 4](https://github.com/user-attachments/assets/863039e0-8c89-464c-aa56-3725fd995e48)
 am totally disturbed by this error in vs code from more than a month , please help me , i have tried to add shortcuts through json file but none worked for me

**Comment by AliceRixte** [comment]
`github.copilot.editor.enableAutoCompletions` does not work anymore. Instead, use ` "github.copilot.enable": { "*": true } ` .

With the [Toggle](https://marketplace.visualstudio.com/items?itemName=rebornix.toggle) extension : 

``` json
{
    "key": "ctrl+i ctrl+t",
    "command": "toggle",
    "when": "editorTextFocus && !editorReadonly && !inSnippetMode && !hasOtherSuggestions && !suggestWidgetVisible",
    "args": {
      "id": "copilot-enabled",
      "overrideWorkspaceSettings": true,
      "value": [
        {
          "github.copilot.enable": { "*": true}
        },
        {
          "github.copilot.enable": { "*": false}
        }
      ]
    }
  },
 ```

**Comment by lorena-davila2025** [comment]
After updating Copilot I'm seeing the option `github.copilot.completitions.toggle` in the `Keyboard Shortcuts` settings with no key binding assigned, so I went and tried `ctrl + p`, and is working really well. 
<img width="1513" alt="Screenshot 2025-07-07 at 9 47 22 AM" src="https://github.com/user-attachments/assets/ee00ddbf-f9a9-41a8-be9d-1de4bc9076c6" />

note: At first I tried the solutions above, but those didn't seem to work anymore in 2025.
